File: ControlGastos/app/src/main/java/com/example/controlgastos/ListItem.java
package com.example.controlgastos;

public class ListItem extends ListNode {

    private double precio;

    public ListItem(String name, double precio) {
        super(name);
        this.precio = precio;
    }

    public void setPrecio(double precio) {
        this.precio = precio;
    }

    @Override
    double getPrecio() {
        return precio;
    }
}

File: ControlGastos/app/src/main/java/com/example/controlgastos/ListItem.kt
package com.example.controlgastos

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
data class ListItem(var name: String, var amount: Double) : ListNode {

    @PrimaryKey
    var id: Int = 0

    override fun getAmount(): Double {
        return amount
    }
}